solutions/src/main/java/ru/sberbank/school/task02/ExtendedFxConversionServiceImpl.java

```diff
@@ -0,0 +1,386 @@
+            BigDecimal curAmount = amount.divide(curPrice, 10, BigDecimal.ROUND_HALF_UP);
+            if (curQuote.equals(findQuote(quoteList, curAmount))) {
+                suitableQuoteList.add(curQuote);
+            }
+        }
+        return suitableQuoteList;
+    }
+    /*
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation,
+                                                Symbol symbol,
+                                                BigDecimal amount,
+                                                Beneficiary beneficiary) {
+        if (amount.compareTo(BigDecimal.ZERO) <= 0) {
+            throw new IllegalArgumentException("Объем не может быть отрицательным или равным нулю");
+        }
+        List<Quote> quoteList = externalQuotesService.getQuotes(symbol);
+        if (quoteList == null || quoteList.isEmpty()) {
+            return Optional.empty();
+        }
+
+        sortQuoteList(quoteList);
+        List<RankedQuote> rankedQuotes = getRankedQuotesList(quoteList, operation);
+
+        int targetQuotePosition;
+        if ((operation == ClientOperation.SELL && beneficiary == Beneficiary.BANK)
+                || (operation == ClientOperation.BUY && beneficiary == Beneficiary.CLIENT)) {
+            targetQuotePosition = getPositionOfFirstConcurrency(rankedQuotes, amount);
+        } else {
+            targetQuotePosition = getPositionOfLastConcurrency(rankedQuotes, amount);
+        }
+
+//        if (operation == ClientOperation.BUY) {
+//            if (beneficiary == Beneficiary.BANK) {
+//                targetQuote = getFirstConcurrency(quoteList, amount, ClientOperation.BUY);
+//            } else {
+//                targetQuote = getLastConcurrency(quoteList, amount, ClientOperation.BUY);
+//            }
+//        } else {
+//            if (beneficiary == Beneficiary.BANK) {
+//                targetQuote = getLastConcurrency(quoteList, amount, ClientOperation.SELL);
+//            } else {
+//                targetQuote = getFirstConcurrency(quoteList, amount, ClientOperation.SELL);
+//            }
+//        }
+
+        if (targetQuotePosition == -1) {
+            return Optional.empty();
+        }
+
+        Quote targetQuote = quoteList.get(targetQuotePosition);
+
+        BigDecimal price = BigDecimal.ONE.divide(
+                operation == ClientOperation.BUY ? targetQuote.getOffer() : targetQuote.getBid(),
+                10, BigDecimal.ROUND_HALF_UP);
+
+        return Optional.of(price);
+    }
+*/
+
+    /**
+     * Возвращает значение цены единицы котируемой валюты для указанного объема.
+     *
+     * @param operation   вид операции
+     * @param symbol      Инструмент
+     * @param amount      Объем
+     * @param delta       допустимое отклонение при отсутствии точного попадания в котируемой валюте
+     * @param beneficiary В чью пользу осуществляется округление
+     * @return Цена для указанного объема
+     */
+    @Override
```

Comment:
Overload methods should not be split. Previous overloaded method located at line &#39;31&#39;.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a2050622567>*